src/sinks/loki.rs

```diff
@@ -243,6 +240,11 @@ impl HttpSink for LokiSink {
             labels = vec![("agent".to_string(), "vector".to_string())]
         }
 
+        let key = PartitionKey {
+            tenant_id,
+            labels: labels.clone(),
```

Comment:
>Is there any way to avoid this clone, 

From what I see, multiple layers would need to be changed to fetch and pass `labels` without cloning it. 

> o cheaply turn the labels into a form

Besides wrapping it in `Arc`, we could just hash them. This partitioning is for performance reason so on hash collision we can send multiple streams in the same request. `LokiBuffer` already has that ability, so just removing simplifications from this PR would be sufficient. This looks like a good way.

Reply:
Right. I was thinking of even just joining them all together into a single string, so it was reduced to one allocation, but a suitable hash would be ideal. However, that is likely better for follow-up work.

Reply:
Good idea, a single string will be cloned in no time.